### PR TITLE
fix: ant-design/ant-design#20999 dropdown hiding issue when select disabled

### PR DESCRIPTION
**Pull request by morenyang**
fix https://github.com/ant-design/ant-design/issues/20999

### CODE REVIEW

**Comment by zombieJ** [comment]
Thanks for PR. Need a test case for this.
* `open` and `disabled` at same time.
* `disabled` after open.
* `open` & `disabled` and then remove `disabled`.

**Comment by morenyang** [comment]
CI failed because of some problem with the definition in `rc-menu`

**Comment by zombieJ** [comment]
CI failed. Please check.

**Comment by morenyang** [comment]
> CI failed. Please check.

Test works well on `rc-menu` at `7.4.30`, but failed at `7.5.4`. Seems ts definition problem in `rc-menu`. @zombieJ could you help to check this issue?

image

image

**Comment by zombieJ** [comment]
Seems not related update recently:
https://github.com/react-component/menu/commits/7.x

If it's OK in working progress, you can `as` it to correct type instead.

**Comment by morenyang** [comment]
Refs: 
https://github.com/react-component/menu/blob/master/src/interface.ts#L7-L12

File: src/Select.tsx [hunk 1]
```diff
@@ -114,6 +114,10 @@ class Select extends React.Component<Partial<ISelectProps>, ISelectState> {
       newState.open = nextProps.open;
     }
 
+    if ('disabled' in nextProps && nextProps.disabled && prevState.open) {
```
Comment (zombieJ) [inline src/Select.tsx]: `'disabled' in nextProps && nextProps.disabled` is same as `nextProps.disabled`

**Pull request merged**